Assets/Decontainment/Singletons/Scripts/PromptSystem.cs

```diff
@@ -0,0 +1,63 @@
+﻿using UnityEngine;
+using System.Collections;
+using TMPro;
+using UnityEngine.UI;
+
+public class PromptSystem : PersistentSingleton<PromptSystem>
```

Comment:
`PersistentSingleton`s are game objects that are not destroyed when a new scene is loaded. The PromptSystem currently acquires a reference to a banner object in `Awake`. When a new scene is loaded, the banner object will be destroyed, but the PromptSystem will persist. If one then tries to call `PromptSystem.Instance.PromptInvalidAction()`, it will call `banner.Activate()`, but banner will be null and the code will break. Therefore, this solution does not work.

I suggest having PromptSystem inherit from `SceneSingleton` then creating a game object with the PromptSystem component in the editor scene. `SceneSingleton`s are destroyed and recreated when a new scene is loaded, so this would fix your issue.